dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/apps/telemetry/ReactiveMessagingTelemetryTest.java

```diff
@@ -0,0 +1,102 @@
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.PropertiesAsset;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.common.ConnectorProperties;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.common.KafkaTestConstants;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.common.KafkaUtils;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.AbstractKafkaTestServlet;
+import com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework.KafkaTestClientProvider;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import io.openliberty.microprofile.reactive.messaging.fat.suite.KafkaTests;
+import io.openliberty.microprofile.reactive.messaging.fat.suite.ReactiveMessagingActions;
+
+/**
+ * Assert what happens when a message from Kafka is nacked
+ */
+@RunWith(FATRunner.class)
+public class ReactiveMessagingTelemetryTest extends FATServletClient {
+
+    private static final String APP_NAME = "ReactiveMessagingTelemetryApp";
+
+    public static final String SERVER_NAME = "RxMessagingServerWithTelemetry";
+
+    @Server(SERVER_NAME)
+    @TestServlet(contextRoot = APP_NAME, servlet = ReactiveMessagingTelemetryTestServlet.class)
+    public static LibertyServer server;
+
+    @ClassRule
+    public static RepeatTests r = ReactiveMessagingActions.repeat(SERVER_NAME,
+                                                                  ReactiveMessagingActions.MP61_RM30,
+                                                                  ReactiveMessagingActions.MP50_RM30,
+                                                                  ReactiveMessagingActions.MP60_RM30);
+
+    @BeforeClass
+    public static void setup() throws Exception {
+
+        PropertiesAsset appConfig = new PropertiesAsset()
+                        .addProperty(AbstractKafkaTestServlet.KAFKA_BOOTSTRAP_PROPERTY, KafkaTests.kafkaContainer.getBootstrapServers())
+                        .include(ConnectorProperties.simpleIncomingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           RmTelemetryProcessingBean.CHANNEL_IN, "ReactiveMessagingTelemetryApp", "Topic"))
+                        .include(ConnectorProperties.simpleOutgoingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           ReactiveMessagingTelemetryTestServlet.EMITTER_CHANNEL, "Topic"))
```

Comment:
* `"Topic"` is a topic name that is likely to clash with other tests. Please pick a name that makes it obvious which test the topic belongs to.
* Use the `APP_NAME` constant instead of `"ReactiveMessagingTelemetryApp"`